src/lib/ng-package/entry-point/write-package.transform.ts

```diff
@@ -35,55 +35,81 @@ export const writePackageTransform = (options: NgPackagrOptions) =>
     ];
 
     if (!ngEntryPoint.isSecondaryEntryPoint) {
-      const assetFiles: string[] = [];
-
       // COPY ASSET FILES TO DESTINATION
       spinner.start('Copying assets');
 
       try {
-        for (const asset of [...ngPackage.assets, 'LICENSE', '**/README.md']) {
-          let assetFullPath = path.join(ngPackage.src, asset);
-
+        type AssetCopyTask = [srcPath: string, destPath: string];
+        const assetCopyTasks: AssetCopyTask[] = [];
+
+        // Parse string assets and add to tasks
+        const assetGlobs = ngPackage.assets
+          .filter((asset): asset is string => typeof asset === 'string')
+          .concat('LICENSE', '**/README.md');
+        for (const assetGlob of assetGlobs) {
```

Comment:
IMHO, the logic here can be simplified and make it easier to follow, if we normalize all entries to globs. Also we need to make sure that the files are within the project root. 

You can take https://github.com/angular/angular-cli/blob/80d486f1592b19a571dfdd847c4811a68679d0cd/packages/angular_devkit/build_angular/src/utils/normalize-asset-patterns.ts#L29-L84 and https://github.com/angular/angular-cli/blob/master/packages/angular_devkit/build_angular/src/utils/copy-assets.ts as an inspiration. 

Also, it would be nice if the logic is moved to a seperate file.

Reply:
Implemented but still in the same file because I'm not sure about how to name the new file.

Reply:
Looks better at first glance but I will review later during the day. 

One thing I noticed (maybe I skipped) is that you missed to add the validation to verify that the input files are within the project root.

Reply:
Yes. The original author @arvindnama requested this feature to allow common assets between multiple projects, which cannot be achived if we limit the input. I accept both ones, so it's up to you.

Reply:
We should definitely limit the input to only copy files within the project root. This is also the same behaviour of Angular CLI.

Reply:
Okay. Btw, Angular CLI allows asset inputs from `node_modules`, maybe we can consider that?